CMakeLists.txt

```diff
@@ -0,0 +1,692 @@
+    "int setcon(char* context)"
+    "0"
+    "selinux/selinux.h"
+    SELINUX_SECURITY_CTX_NO_CONST
+  )
+  if (SELINUX_SECURITY_CTX_NO_CONST)
+    message(STATUS "security_context_t is char*")
+    set(KLEE_SELINUX_CTX_CONST " ")
+  else()
+    check_prototype_definition(setcon
+      "int setcon(const char* context)"
+      "0"
+      "selinux/selinux.h"
+      SELINUX_SECURITY_CTX_WITH_CONST
+    )
+    if (SELINUX_SECURITY_CTX_WITH_CONST)
+      message(STATUS "security_context_t is const char*")
+      set(KLEE_SELINUX_CTX_CONST "const")
+    else()
+      message(FATAL_ERROR "Failed to determine function signature for \"setcon\"")
+    endif()
+  endif()
+else()
+  message(STATUS "SELinux support disabled")
+  set(HAVE_SELINUX 0)
+endif()
+
+################################################################################
+# KLEE runtime support
+################################################################################
+# This is set here and not in `runtime` because `config.h` needs to be generated.
+
+
+set(available_klee_runtime_build_types
+  "Release"
+  "Release+Asserts"
+  "Debug"
+  "Debug+Asserts"
+)
+if (NOT KLEE_RUNTIME_BUILD_TYPE)
+  message(STATUS "KLEE_RUNTIME_BUILD_TYPE is not set. Setting default")
+  message(STATUS "The available runtime build types are: ${available_klee_runtime_build_types}")
+  set(KLEE_RUNTIME_BUILD_TYPE "Release+Asserts" CACHE String
+    "Options are ${available_klee_runtime_build_types}"
+    FORCE)
+    # Provide drop down menu options in cmake-gui
+    set_property(CACHE
+      KLEE_RUNTIME_BUILD_TYPE
+      PROPERTY STRINGS ${available_klee_runtime_build_types})
+endif()
+message(STATUS "KLEE_RUNTIME_BUILD_TYPE: ${KLEE_RUNTIME_BUILD_TYPE}")
+set(RUNTIME_CONFIGURATION "\"${KLEE_RUNTIME_BUILD_TYPE}\"") # For config.h
+
+set(KLEE_INSTALL_RUNTIMR_DIR "${CMAKE_INSTALL_FULL_LIBDIR}/klee/runtime")
+# FIXME: Move this into `config.h`
+list(APPEND KLEE_COMPONENT_CXX_DEFINES
+  -DKLEE_DIR=\"${CMAKE_BINARY_DIR}\"
+  -DKLEE_INSTALL_BIN_DIR=\"${CMAKE_INSTALL_FULL_BINDIR}\"
+  -DKLEE_INSTALL_RUNTIME_DIR=\"${KLEE_INSTALL_RUNTIMR_DIR}\")
+
+# Location where KLEE will look for the built runtimes by default.
+set(KLEE_RUNTIME_DIRECTORY "${CMAKE_BINARY_DIR}/${KLEE_RUNTIME_BUILD_TYPE}/lib")
+
+################################################################################
+# KLEE POSIX Runtime Support
+################################################################################
+option(ENABLE_POSIX_RUNTIME "Enable KLEE's POSIX runtime" OFF)
+if (ENABLE_POSIX_RUNTIME)
+  message(STATUS "POSIX runtime enabled")
+else()
+  message(STATUS "POSIX runtime disabled")
+endif()
+
+################################################################################
+# KLEE uclibc support
+################################################################################
+option(ENABLE_KLEE_UCLIBC "Enable support for klee-uclibc" OFF)
+if (ENABLE_KLEE_UCLIBC)
+  message(STATUS "klee-uclibc support enabled")
```

Comment:
We probably should warn. I actually found that our testing is kind of broken here too. If we configure with the posix runtime enabled but without klee-uclibc then we will try to run a bunch of integration tests that pass `-libc=uclibc` to KLEE and so fail. The CMake build system currently emits an author warning for this (posix runtime enabled but klee-uclibc disabled) problem when it is configured this way to remind us to fix this.